react-components/src/apps/admin/AllStats/AppStatsComponents/datePicker.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Used for Start Date & End Date
+ */
+
+import React from "react";
+import {
+    MuiPickersUtilsProvider,
+    KeyboardDatePicker,
+} from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
+import PropTypes from "prop-types";
+
+function DatePicker(props) {
+    const id = props.id,
+        label = props.label,
+        value = props.selectedDate,
+        dateChange = props.dateChange;
+
+    return (
+        <div className={props.className}>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                <KeyboardDatePicker
+                    disableToolbar
+                    variant="inline"
+                    format="MM/dd/yyyy"
+                    id={id}
+                    label={label}
+                    value={value}
+                    onChange={dateChange}
+                    KeyboardButtonProps={{
+                        "aria-label": "change date",
+                    }}
+                />
+            </MuiPickersUtilsProvider>
+        </div>
+    );
+}
+
+DatePicker.propTypes = {
```

Comment:
Sorry I didn't notice this before, but this `DatePicker.propTypes` is declaring that `DatePicker` requires the `presenter` and selected date props. The actual `DatePicker` function above is only using `dateChange` and `selectedDate` from its `props` though.

It looks like these Prop Types should actually be assigned to `AllStats.propTypes`.

Reply:
How come it didn't show any errors when it was DatePicker.propTypes?

Reply:
There should have been some warnings in the console about failed prop types.

I also seem to remember that when you're running Storybook, sometimes warnings will only display once at the start, but not again after code changes and the page auto-reloads. I think you can get the warnings to display again if you manually refresh the browser tab, or by restarting Storybook.